torch/_dynamo/utils.py

```diff
@@ -1013,30 +1013,29 @@ def record_compilation_metrics(
     exc_type: Optional[Type[BaseException]],
     exc_value: Optional[BaseException],
 ):
-    def us_to_s(field):
-        metric = metrics.get(field, None)
+    def us_to_s(m: Dict[str, Any], field: str) -> Optional[float]:
+        metric = m.get(field, None)
         return metric / 1e6 if metric is not None else None
 
-    def us_to_ms(field):
-        metric = metrics.get(field, None)
+    def us_to_ms(m: Dict[str, Any], field: str) -> Optional[int]:
+        metric = m.get(field, None)
         return metric // 1000 if metric is not None else None
 
-    def _convert_collection_to_str(field: str) -> Optional[str]:
+    def collection_to_str(m: Dict[str, Any], field: str) -> Optional[str]:
         def safe_str(item: Any) -> str:
             try:
                 return str(item)
             except Exception:
-                return str(None)
+                return "<unknown>"
 
-        metric = metrics.get(field, None)
+        metric = m.get(field, None)
         if metric is None:
             return None
 
-        # Remove this field (list/set) from metrics to avoid clashes
-        del metrics[field]
-        if not isinstance(metric, set) and not isinstance(metric, list):
-            return None
-        return ",".join(safe_str(item) for item in metric)
+        if not isinstance(metric, (set, list)):
+            return "<unknown>"
+
+        return ",".join(safe_str(item) for item in sorted(metric))
```

Comment:
Note the "sorted". I don't know of any problems currently, just given the in-person discussion today I thought I may as well change it since I was already touching this code.